assume/world.py

```diff
@@ -264,28 +274,32 @@ def creator(container):
                 market_config=market_config,
             )
 
-        # add making price forecast
-        forecast_role = ForecastProvider(
-            config["markets_config"].items(),
-            fuel_prices_df,
-            fuel_prices_df["co2"],
-            vre_cf_df,
-            powerplant_units,
-            demand_df,
-        )
-        forecast_agent = RoleAgent(self.container, suggested_aid="forecast_agent_1")
-        forecast_agent.add_role(forecast_role)
-
-        # write forecast
-        self.price_forecast = forecast_role.calculate_price_forecast(
-            "EOM",
-            path,
-            demand_df,
-            vre_cf_df,
-            powerplant_units,
-            fuel_prices_df,
-            fuel_prices_df["co2"],
-        )
+        # add forecast providers for each market
+        self.logger.info("Adding forecast providers")
+        temp = pd.DataFrame()
+        for market_id, market_config in self.markets.items():
+            if price_forecast_df is not None and market_id in price_forecast_df.columns:
+                market_price_forecast = price_forecast_df[market_id]
+            else:
+                market_price_forecast = None
+
+            forecast_provider = ForecastProvider(
+                market_id=market_id,
+                price_forecast_df=market_price_forecast,
+                fuel_prices_df=fuel_prices_df,
+                vre_cf_df=vre_cf_df,
+                powerplants=powerplant_units,
+                demand_df=demand_df[f"demand_{market_id}"],
+            )
+            forecast_agent = RoleAgent(
+                self.container, suggested_aid=f"forecast_agent_{market_id}"
+            )
+            forecast_agent.add_role(forecast_provider)
+            self.forecast_providers[market_id] = forecast_provider
+
+            temp[market_id] = forecast_provider.price_forecast_df
+
+        temp.to_csv(f"{path}/price_forecasts.csv", index=True)
```

Comment:
why are we keeping this additional file?

Reply:
so it won't waste time calculating it again and again each time you run a simulation